server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferManager.java

```diff
@@ -269,6 +269,15 @@ public boolean downloadTranslog(String primaryTerm, String generation, Path loca
         return true;
     }
 
+    /**
+     * markFileAsDownloaded marks a file as already downloaded in the file transfer tracker.
+     * This is needed because we want to ensure that even if the file is skipped from being downloaded,
+     * TranslogTransferManager knows about it and does not re-upload the same to remote store.
+     * */
+    public void markFileAsDownloaded(String filename) {
+        fileTransferTracker.add(filename, true);
+    }
+
```

Comment:
I am assuming file names bear index/shard id discriminators?

Reply:
Although the translog and checkpoint file names won't include a shard identifier, the `TranslogTransferManager` instance responsible for marking the files as downloaded will be specifically created for each shard. The translog and checkpoint files will keep the existing naming convention. 

Example-
```
translog-21.tlog
translog-21.ckp
```

Ref- https://github.com/rawahars/OpenSearch/blob/ce18c9bc945e59aa279f3083a455c18244697f8f/server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java#L197

Reply:
Git it `TranslogTransferManager` is per shardId